modules/social_features/social_group/src/Plugin/Action/SocialSendEmail.php

```diff
@@ -20,18 +36,304 @@
  */
 class SocialSendEmail extends SocialSendEmailBase {
 
+
+  use ViewsBulkOperationsFormTrait;
```

Comment:
I didn't notice this before, but if I read the description for this trait, then it really shouldn't be used in Action's. What do we need it for?

Reply:
This trait needed for calling `$this->getListItem()` and `$this->setTempstoreData()`. 
You can see that `$this->setTempstoreData()` is a bit complex to copy-past.

Reply:
We should find a follow-up to get rid of that trait. I think it shows that we need to dive too deeply into VBO internals here, which is likely to break in future updates.

However, at least in the short term I don't have enough knowledge of VBO to give you an alternative so I guess we stick with it for now.